### ISSUE TITLE
Problems with matching strings

### ISSUE THREAD

**Issue by ulrikstrid** [facebookexperimental/reason-native#121]
I'm writing tests for a function that throws a string that is the combination of keys from a Hashtable. I'm not sure how deterministic the order is so I figured I'll test each key that I know will be there. This works fine for every key except the last one.

Tests that reproduces the error:
```ocaml
describe("Safe functionality", utils => {
    utils.test("Minimal repro", ({expect}) => {
    let myString = "TEST2, TEST, TEST3";

    expect.string(myString).toMatch("TEST");
    expect.string(myString).toMatch("TEST2");
    expect.string(myString).toMatch("TEST3");
  });

  utils.test("Minimal repro 2", ({expect}) => {
    let myString = "TEST, TEST2, TEST3";

    expect.string(myString).toMatch("TEST");
    expect.string(myString).toMatch("TEST2");
    expect.string(myString).toMatch("TEST3");
  });
});
```
And this is the output of those tests:
```
  • Safe functionality › Minimal repro

    expect.string(received).toMatch(expected)
    
    Expected value to match:
      "TEST3"
    Received:
      "TEST2, TEST, TEST3"

  • Safe functionality › Minimal repro 2

    expect.string(received).toMatch(expected)
    
    Expected value to match:
      "TEST2"
    Received:
      "TEST, TEST2, TEST3"
```

**Comment by bandersongit** [comment]
This current behavior is expected as we are using Str.string_match, however I can definitely see this being confusing, I imagine what we want is the behavior of Str.string_partial_match (and there is some ongoing discussion of if we want to use Str at all)

**Comment by ulrikstrid** [comment]
As I said on discord I got past it by changing my test, but this seems like a unintuitive behaviour without knowing about the internals. Feel free to close it if it's not relevant

**Comment by kyldvs** [comment]
@ulrikstrid Is there something else you would call `toMatch` to communicate that the pattern needs to match the entire string?

**Comment by bandersongit** [comment]
I was actually thinking about it backwards, now that I think about it some more (I think I was thinking about what expect.string.toContain to do), I think the current behavior is correct and at least consistent with Jest and the behavior of regex matching in general.

I could see changing the argument to toMatch to be a custom Regexp variant to make it more clear, I don't think we want to tie it to a particular library.

Given this, I am going to close this now, but I do expect to stop using Str in the near future, and I think expect.string.toContain would be a nice enhancement.

**Comment by bandersongit** [comment]
Reopening as @kyldvs jumped in at the same time :p

**Comment by ulrikstrid** [comment]
@kyldvs The problem is that toMatch now matches on substrings as my repro shows. Either way it doesn't do what we expect.

@bandersongit I think toContain would be perfect for my usecase.

**Comment by kyldvs** [comment]
Oh I see, it is weird that it matches just a prefix but does not try to match the entire string. We probably should switch this to https://caml.inria.fr/pub/docs/manual-ocaml/libref/Str.html#VALsearch_forward, I think that will be aligned with how Jest's `toMatch` works: https://jestjs.io/docs/en/expect#tomatchregexporstring

**Comment by bandersongit** [comment]
This change was included in Rely 1.3.1
